collector/softnet_linux.go

```diff
@@ -76,8 +106,8 @@ func (c *softnetCollector) Update(ch chan<- prometheus.Metric) error {
 		return fmt.Errorf("could not get softnet statistics: %w", err)
 	}
 
-	for cpuNumber, cpuStats := range stats {
-		cpu := strconv.Itoa(cpuNumber)
+	for _, cpuStats := range stats {
+		cpu := strconv.FormatUint(uint64(cpuStats.Index), 10)
```

Comment:
Due to the differing columns between kernel versions, I don't think we can rely on `Index` being valid.

I think we may have made a minor mistake in the procfs change, since Index is an optional column, we should set it based on the range index.

Reply:
we could use cpuStats.Index if cpuStats.Width >= 13, if cpuStats.Width <= 13 we can fallback to cpuNumber, what do you think ?